src/emulator/apphosting/config.spec.ts

```diff
@@ -9,76 +8,125 @@ import * as configImport from "../../apphosting/config";
 import { AppHostingYamlConfig } from "../../apphosting/yaml";
 
 describe("environments", () => {
-  let pathExistsStub: sinon.SinonStub;
   let joinStub: sinon.SinonStub;
   let loggerStub: sinon.SinonStub;
   let loadAppHostingYamlStub: sinon.SinonStub;
-  let discoverConfigsAtBackendRoot: sinon.SinonStub;
+  let listAppHostingFilesInPathStub: sinon.SinonStub;
+
+  // Configs used for stubs
+  const apphostingYamlConfigOne = AppHostingYamlConfig.empty();
+  apphostingYamlConfigOne.addEnvironmentVariable({
+    variable: "randomEnvOne",
+    value: "envOne",
+  });
+  apphostingYamlConfigOne.addEnvironmentVariable({
+    variable: "randomEnvTwo",
+    value: "envTwo",
+  });
+  apphostingYamlConfigOne.addEnvironmentVariable({
+    variable: "randomEnvThree",
+    value: "envThree",
+  });
+  apphostingYamlConfigOne.addSecret({ variable: "randomSecretOne", secret: "secretOne" });
+  apphostingYamlConfigOne.addSecret({ variable: "randomSecretTwo", secret: "secretTwo" });
+  apphostingYamlConfigOne.addSecret({ variable: "randomSecretThree", secret: "secretThree" });
+
+  const apphostingYamlConfigTwo = AppHostingYamlConfig.empty();
+  apphostingYamlConfigTwo.addEnvironmentVariable({
+    variable: "randomEnvOne",
+    value: "envOne",
+  });
+  apphostingYamlConfigTwo.addEnvironmentVariable({
+    variable: "randomEnvTwo",
+    value: "blah",
```

Comment:
nit* I think the test case would be a lot clearer if these override markers are self descriptive, e.g.

"value: "THIS IS FROM SECOND CONFIG""